src/Microsoft.TemplateSearch.Common/BlobStorageTemplateInfo.cs

```diff
@@ -158,70 +110,160 @@ IReadOnlyList<ITemplateParameter> ITemplateInfo.Parameters
+                                    DefaultValue = tagObj.TryGetValue(nameof(BlobTemplateParameter.DefaultValue), StringComparison.OrdinalIgnoreCase, out JToken defValue) ? defValue.ToString() : null
+                                });
+                        }
+                        tags[item.Name.ToString()] = tagObj.TryGetValue(nameof(BlobTemplateParameter.DefaultValue), StringComparison.OrdinalIgnoreCase, out JToken defaultValue) ? defaultValue.ToString() : "";
+                    }
+                }
+            }
 
-        [JsonProperty]
-        public string? Description { get; private set; }
+            if (!readParameters && entry.TryGetValue("cacheParameters", StringComparison.OrdinalIgnoreCase, out JToken cacheParameters) && cacheParameters is JObject cacheParametersObject)
+            {
+                foreach (JProperty item in cacheParametersObject.Properties())
+                {
+                    JObject paramObj = (JObject)item.Value;
+                    if (paramObj == null)
+                    {
+                        continue;
+                    }
+                    string dataType = paramObj.TryGetValue(nameof(BlobTemplateParameter.DataType), StringComparison.OrdinalIgnoreCase, out JToken dt) ? dt.ToString() : "";
+                    templateParameters.Add(
+                        new BlobTemplateParameter(item.Name.ToString(), "parameter", dataType)
+                        {
+                            Description = paramObj.TryGetValue(nameof(BlobTemplateParameter.Description), StringComparison.OrdinalIgnoreCase, out JToken description) ? description.ToString() : null,
+                            DefaultValue = paramObj.TryGetValue(nameof(BlobTemplateParameter.DefaultValue), StringComparison.OrdinalIgnoreCase, out JToken defValue) ? defValue.ToString() : null
+                        });
+                }
+            }
+            info.Parameters = templateParameters;
+            return info;
 
-        [JsonIgnore]
-        public IReadOnlyDictionary<string, ParameterChoice> Choices => _choices.ToDictionary(kvp => kvp.Key, kvp => new ParameterChoice(null, kvp.Value));
+        }
 
-        [JsonProperty]
-        public string? DefaultValue { get; private set; }
-    }
+        internal class BaselineCacheInfo : IBaselineInfo
+        {
+            [JsonProperty]
+            public string? Description { get; private set; }
 
-    internal class CacheParameter : ICacheParameter
-    {
-        [JsonProperty]
-        public string? DataType { get; private set; }
+            [JsonProperty]
+            public IReadOnlyDictionary<string, string> DefaultOverrides { get; private set; } = new Dictionary<string, string>();
+        }
 
-        [JsonProperty]
-        public string? DefaultValue { get; private set; }
+        internal class BlobTemplateParameter : ITemplateParameter
```

Comment:
I think it would be better if we had a folder for all "BlobStorageTemplate" related types where each type is a separate file.

Reply:
this class became private.